packages/components/Shimmer/src/Shimmer.tsx

```diff
@@ -0,0 +1,99 @@
+/** @jsx withSlots */
+import { compose, UseSlots, buildProps, mergeProps, withSlots } from '@fluentui-react-native/framework';
+import { requireNativeComponent } from 'react-native';
+import { IViewProps } from '@fluentui-react-native/adapters';
+import * as React from 'react';
+
+const shimmerName = 'Shimmer';
+
+const NativeShimmerView = requireNativeComponent('MSFShimmerView');
+
+/**
+ * Appearance of the views that are shimmered
+ */
+export interface IShimmerViewAppearance {
+  /**
+   * Color to tint the shimmer boxes. Defaults to the Fabric default color.
+   */
+  tintColor: number;
+
+  /**
+   * Corner radius on each view.
+   */
+  cornerRadius: number;
+
+  /**
+   * Corner radius on each UILabel. Set to < 0 to disable and use default `cornerRadius`.
+   */
+  labelCornerRadius: number;
+
+  /**
+   * usesTextHeightForLabels: True to enable shimmers to auto-adjust to font height for a UILabel -- this
+   * will more accurately reflect the text in the label rect rather than using the bounding box. `labelHeight`
+   * will take precedence over this property.
+   */
+  usesTextHeightForLabels: boolean;
+
+  /**
+   * If greater than 0, a fixed height to use for all UILabels. This will take precedence
+   * over `usesTextHeightForLabels`. Set to less than 0 to disable.
+   */
+  labelHeight: number;
+}
+
+/**
+ * Appearance of the shimmer itself (the animation appearance)
+ */
+export interface IShimmerAppearance {
+  alpha: number;
+  width: number;
+
+  /**
+   * Angle of the direction of the gradient, in radian. 0 means horizontal, Pi/2 means vertical.
+   */
+  angle: number;
+
+  /**
+   * Speed of the animation, in point/seconds.
+   */
+  speed: number;
+
+  /**
+   * Delay between the end of a shimmering animation and the beginning of the next one, in seconds
+   */
+  delay: number;
+}
+
+export interface ShimmerTokens {}
+
+/**
+ * ViewProps props, based off of the standard react-native ViewProps with some new extensions
+ */
+export type ShimmerProps<TBase = IViewProps> = TBase & {
+  /**
+   * Appearance of the views that are shimmered. If not passed, the default will be used.
+   */
+  appearance?: IShimmerViewAppearance;
+
+  /**
+   * Appearance of the shimmer itself (the animation appearance)
+   */
+  shimmerAppearance?: IShimmerAppearance;
+};
+
+interface ShimmerType {
+  props: ShimmerProps;
+  slotProps: { root: ShimmerProps };
```

Comment:
Note it seems like the ShimmerProps interface is the internal interface for the native component, which may make sense. The externally facing interface can be one that is more standard for react/react-native.

Reply:
This means that ShimmerProps might be renamed as NativeShimmerProps (and set in slotProps), then have ShimmerProps be the outward facing interface to RN consumers. Then your component can transform them as appropriate.